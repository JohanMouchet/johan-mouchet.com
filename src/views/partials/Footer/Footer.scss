@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.p-footer {
  margin-top: auto;
  padding: t-vertical-rhythm(1.2) 0;
  background: map-get($s-backgrounds, dots),
    url("#{map-get($s-paths, theme)}/footer.jpg") right center/cover no-repeat
      $s-color-primary;
  letter-spacing: 0.1em;
  font-family: $s-font-family-mono;
  font-size: t-modular-scale(-1);
  text-shadow: 1px 1px 1px rgba($s-color-black, 0.3);
  color: $s-color-white;
  position: relative;

  &::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background-image: linear-gradient(
      rgba($s-color-primary-darker, 0.9),
      transparent
    );
  }

  & > * {
    position: relative;
  }

  &__copyright {
    margin-bottom: 0;
    text-align: center;

    &-separator {
      display: none;
    }

    @media (min-width: map-get($s-breakpoints, sm)) {
      &-separator {
        display: inline-block;
      }

      br {
        display: none;
      }
    }
  }

  &__cocorico {
    /* stylelint-disable color-no-hex */
    $s-color-bleu: #2b6cb0;
    $s-color-rouge: #e53e3e;
    /* stylelint-enable color-no-hex */

    position: relative;
    background-color: $s-color-white;
    text-shadow: none;
    outline: 0;
    transition: background-color 0.2s ease-out, text-shadow 0.2s ease-out;

    &::before,
    &::after {
      content: "";
      position: absolute;
      top: 0;
      height: 100%;
      width: 33%;
      transition: background-color 0.2s ease-out, left 0.2s ease-out,
        right 0.2s ease-out;
    }

    &::before {
      background: $s-color-bleu;
      left: 0;
    }

    &::after {
      background: $s-color-rouge;
      right: 0;
    }

    &:hover,
    &:focus {
      background-color: transparent;
      text-shadow: inherit;

      &::before {
        background-color: transparent;
        left: -33%;
      }

      &::after {
        background-color: transparent;
        right: -33%;
      }
    }
  }

  &__page-top {
    display: inline-block;
    line-height: 0;
    padding: 0.3em;
    border: 2px solid $s-color-white;
    transition: opacity 0.2s ease-out, transform 0.1s ease-out;

    @include t-wekkit-hdpi-small-dimension("border-width", 2);

    svg {
      transform: rotate(180deg);
      height: t-modular-scale(2);
      width: t-modular-scale(2);
      fill: $s-color-white;
    }

    &:hover,
    &:focus {
      opacity: 0.5;
    }

    &:active {
      transform: translateY(2px);
    }
  }
}
