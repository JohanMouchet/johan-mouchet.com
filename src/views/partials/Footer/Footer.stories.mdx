import { Canvas, Meta, Story } from "@storybook/addon-docs";
import { text } from "@storybook/addon-knobs";
import Footer from ".";

<Meta title="Partials/Footer" component={Footer} />

# Footer

<Canvas>
  <Story name="default">
    <Footer
      version={text("version")}
      copyright={text("copyright")}
      topButtonLabel={text("topButtonLabel")}
    />
  </Story>
</Canvas>
