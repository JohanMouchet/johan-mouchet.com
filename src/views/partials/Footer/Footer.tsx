import * as React from "react";
import { VERSIONS } from "constants/versions";
import { PROFILE } from "constants/profile";
import { IconChevronDown } from "views/objects";
import "./Footer.scss";

type Props = {
  version: string;
  copyright: string;
  topButtonTitle: string;
};

const Footer = ({
  version = VERSIONS.site,
  /* html*/
  copyright = `${PROFILE.firstName} ${PROFILE.lastName.toUpperCase()}`,
  /* html*/
  topButtonTitle = `Get to the top`,
}: Props) => {
  const year = new Date().getFullYear();

  return (
    <footer className="p-footer">
      <div className="container">
        <div className="grid grid--yCenter-@xs grid--noWrap-@xs">
          <div className="cell cell--0-@xs">
            <span className="p-footer__version">{version}</span>
          </div>
          <div className="cell cell--@xs">
            <p className="p-footer__copyright">
              {year} &copy; {copyright}{" "}
              <span className="p-footer__copyright-separator">|</span>
              <br /> Made in{" "}
              <span className="p-footer__cocorico" tabIndex="0">
                France
              </span>
            </p>
          </div>
          <div className="cell cell--0-@xs">
            <a
              className="p-footer__page-top"
              href="#page-top"
              aria-label={topButtonTitle}
            >
              <IconChevronDown />
            </a>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
