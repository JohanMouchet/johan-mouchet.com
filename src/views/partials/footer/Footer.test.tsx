import { render } from "@testing-library/react";
import { Footer } from "./Footer";

it("Renders <Footer /> without crashing", () => {
  expect(
    render(
      <Footer
        version="Version"
        copyright="Copyright"
        topButtonLabel="Top Button Label"
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
