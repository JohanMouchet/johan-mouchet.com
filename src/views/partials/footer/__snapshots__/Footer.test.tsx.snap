// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Footer /> without crashing 1`] = `
<footer
  class="p-footer"
>
  <div
    class="container"
  >
    <div
      class="grid grid-y-center grid-no-wrap"
    >
      <div
        class="cell cell-shrink"
      >
        <span
          class="p-footer__version"
        >
          Version
        </span>
      </div>
      <div
        class="cell cell-grow"
      >
        <p
          class="p-footer__copyright"
        >
          2023
           Â© 
          Copyright
           
          <span
            class="p-footer__copyright-separator"
          >
            |
          </span>
          <br />
           Made in
           
          <span
            class="p-footer__cocorico"
            tabindex="0"
          >
            France
          </span>
        </p>
      </div>
      <div
        class="cell cell-shrink"
      >
        <a
          aria-label="Top Button Label"
          class="p-footer__page-top"
          href="#page-top"
        >
          <svg />
        </a>
      </div>
    </div>
  </div>
</footer>
`;
